### ISSUE TITLE
Quantized tflite model always predict class as zero for any images 

### ISSUE THREAD

**Issue by lychenpan** [tensorflow/models#5749]
Generated quantized tflite model predicts all the same.  
------------------------

### System information
- **What is the top-level directory of the model you are using**:  tensorflow/models/research/  
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:  
- **TensorFlow installed from (source or binary)**: keras-docker
- **TensorFlow version (use command below)**:   1.11.0
- **Bazel version (if compiling from source)**:  
- **CUDA/cuDNN version**:  
- **GPU model and memory**: 11G
- **Exact command to reproduce**:  

### Describe the problem
I want to use quantized model to detect object on mobile.  1. I fine-tune a model from ssdlite-mobile-v2-coco model, (it has no problem when testing it on PC)  2. I use "object_detection/export_tflite_ssd_graph.py "  to export  *.pb and *.pbtxt from the checkpoint folder  3. use "tflite_convert" commnad to convert *pb file to tflite(both float and quantized are tried). 
When i covert *pb to float tflite model, the generated model works well on mobile;  but when convert it to quantized tflite model,  it's class prediction are always zero, and their score is 0.5,  for any image input. 

### Source code / logs
1.  script to export *pb file from check point folder.

export PYTHONPATH=$PYTHONPATH:`pwd`:`pwd`/slim
CUDA_VISIBLE_DEVICES=1 \
python object_detection/export_tflite_ssd_graph.py \
--pipeline_config_path=train/carpart_base/pipeline.config \
--trained_checkpoint_prefix=train/carpart_base/model.ckpt-600000 \
--output_directory=output2 \
--add_postprocessing_op=true

2. script to covert pb file to tflite model
tflite_convert \
--output_file=/src/workspace/models/research/output3/float1.tflite \
--graph_def_file=/src/workspace/models/research/output3/tflite_graph.pb \
--input_format=TENSORFLOW_GRAPHDEF \
--output_format=TFLITE \
--input_shapes=1,300,300,3 \
--input_arrays=normalized_input_image_tensor \
--output_arrays='TFLite_Detection_PostProcess','TFLite_Detection_PostProcess:1','TFLite_Detection_PostProcess:2','TFLite_Detection_PostProcess:3' \
--input_type=FLOAT \
--inference_type=FLOAT \
--change_concat_input_ranges=false \
--allow_custom_ops
Above float model works well on mobile.

tflite_convert \
--output_file=/src/workspace/models/research/output2/quant4.tflite \
--graph_def_file=/src/workspace/models/research/output2/tflite_graph.pb \
--input_format=TENSORFLOW_GRAPHDEF \
--output_format=TFLITE \
--input_shapes=1,300,300,3 \
--input_arrays=normalized_input_image_tensor \
--output_arrays='raw_outputs/box_encodings','raw_outputs/class_predictions' \
--input_type=QUANTIZED_UINT8 \
--inference_type=QUANTIZED_UINT8 \
--change_concat_input_ranges=false \
--mean_values=0 \
--std_dev_values=1 \
--default_ranges_min=0 \
--default_ranges_max=6 \
This quantized model predicts all the same for class and class score.  And i have tried different mean/std(128,128), which not works either. 
Can anyone tell me about the error or how to calculate the mean/std parameters?
Thanks!

**Comment by lychenpan** [comment]
Take some comment here if some one have same problem.
The reason is that, tflite don't support some op of my training model: ssdlite-mobilenet v2.   In this situation, tflite_convert will use a fake op(I guess it will output zero values for any input) to replace the unsupported op without any warnings and errors.  
Solution:  When training or finetuning models, Add a graphrewriter{} with quantization config, then convert the generated model to tflite will work well.

**Comment by zehualiu035** [comment]
Hi, I meet the same issue with same model, can you please provide some detail or sample core for the solution? Thank you

**Comment by anis-13** [comment]
hey @lychenpan @zehualiu035  ! i'm facing the same probleme ! i run prediction on the tflite converted model on images and that's an overview of results that i got !
[[ 0.24844283 -0.05162451  0.4484877   0.55496323]
 [ 0.5988011   0.3000673   0.89830893  0.79695296]
 [ 0.65304977  0.30025536  0.79501706  0.64705396]
 [ 0.02581225 -0.10002244  0.27102855  0.39686322]
 [ 0.46978283  0.212951    1.0376521   0.7808203 ]
 [ 0.49301383  0.11615509  0.77694845  0.2581224 ]
 [ 0.33878565  0.1387408   0.7582346   0.55818975]
 [ 0.27102852  0.5678693   0.41299587  0.851804  ]
 [ 0.84857744  0.34846523  1.0486224   0.955053  ]
 [ 0.76468766  0.6711183   0.8614836   0.76791424]]

**Comment by zehualiu035** [comment]
@anis-13 try   
  detected_boxes = interpreter.get_tensor(output_details[0]['index']) * height
  detected_classes = interpreter.get_tensor(output_details[1]['index'])
  detected_scores = interpreter.get_tensor(output_details[2]['index'])
  num_boxes = interpreter.get_tensor(output_details[3]['index'])

**Comment by anis-13** [comment]
@zehualiu035 thank you for answer ! i have managed to fix the probleme! your solution was the right one

**Comment by Davari393** [comment]
hi zehualiu035.i have this problem too.but when i write these:
detected_boxes = interpreter.get_tensor(output_details[0]['index']) * height
detected_classes = interpreter.get_tensor(output_details[1]['index'])
detected_scores = interpreter.get_tensor(output_details[2]['index'])
num_boxes = interpreter.get_tensor(output_details[3]['index'])

error appears that list is out of range.do you know why?

**Comment by anis-13** [comment]
@Davari393 check the size of your inputs! the size of the images should be the same as the input_tensor !

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.

**Comment by imaneBYD2020** [comment]
hello plz i got the same issue when i use a tflite with categorical classes

**Comment by imaneBYD2020** [comment]
any help plz
